pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -546,7 +546,18 @@ def is_using_torchelastic(self) -> bool:
         )
         return TorchElasticEnvironment.is_using_torchelastic()
 
+    def _validate_precision_type(self) -> None:
+        """
+        Ensures that the set precision type by the user is valid.
+        """
+        valid_types = (16, 32, 64, "bf16")
```

Comment:
Maybe create a `PrecisionType` Enum?

Reply:
I think we should support bfloat16 too as user might try this option first coming from PyTorch documentation after reading updates.

Reply:
Agreed!

Reply:
And I think we should also add `precision='amp'` and deprecate `precision=16` for now and as soon as we remove it, we can add true half precision training instead of amp.